### PR TITLE
Action: Add Basic and GitHub token Authentication

### PR DESCRIPTION
**Pull request by ethan92429**
This will add Basic Authentication to the fetch.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -21,9 +21,18 @@ inputs:
...
+    required: false
+  password:
+    description: 'Used in `Basic` authentication'
+    required: false
 outputs:
   file-path:
     description: 'The full path to the downloaded file'
 runs:
-  using: 'node16'
+  using: 'node12'
```
Comment (pzhlkj6612) [inline action.yml:37]: Hi!  
Maybe this is an unnecessary modification?

Comment (ethan92429) [reply action.yml:37]: I'm fine changing it back - It is just that making it node12 makes it compatible with https://github.com/nektos/act out of the box.

Comment (ethan92429) [reply action.yml:37]: And there are no downsides AFAICT...

Comment (pzhlkj6612) [reply action.yml:37]: > It is just that making it node12 makes it compatible with https://github.com/nektos/act out of the box.

Oh, I didn't know "nektos/act", thank you.  
And I've found that the node16 support has been added in nektos/act#922 , did you try that?

> And there are no downsides AFAICT...

Well, yes, but it's not relevant to this PR. IMHO, to keep git-blame clean, we need to make changes as small as possible.

Comment (ethan92429) [reply action.yml:37]: Makes total sense. Works for me! Updating now. Thanks!

**Comment by ethan92429** [comment]
OK - should be GTG. Happy to change anything else as needed.

**Comment by carlosperate** [comment]
Hi @ethan92429, first of all I'd like to thank you for your contribution, it is very much appreciated!

The patch looks good, but I have concerns about the general idea of maintaining an action that deals with secrets. I'm not versed enough in this area to be confident I could ensure a tool like this could not offer a way for malicious actors to steal this data.
 
Even if users of this action used GitHub secrets, somebody could raise a PR to download a file from their own server and log this. To be fair, GitHub has recently added PR approvals, which places the responsability of checking the PR on the maintainers before approving the CI, and tecnically a malicious actor could do a similar kind of thing with wget or curl.
All that being said, I don't think I'd be able to provide the level of security (some) users might expect when an action provides this kind of feature.

**Comment by ethan92429** [comment]
Hi @carlosperate,

No worries! I'll just keep using my fork 😄 Luckily GH makes it easy enough to do that. Thanks for the action in the first place!

**Comment by pzhlkj6612** [comment]
A note to @ethan92429 : to work with a fork without telling the subscribers of the original repo what commits you added, you may need to close this PR, or change the working branch in your fork (now is "master").

**Comment by ethan92429** [comment]
The maintainers opted to not incorporate any authentication into this action, so anyone searching for a solution with authentication are welcome to check out the referenced fork. Please be aware that I am not planning to maintain the fork.

**Pull request closed without merging**